src/Display.java

```diff
@@ -0,0 +1,43 @@
+public class Display {
+
+    public static void showDataAnimal(GenericContainer<? extends Animal> container) {
+        System.out.println(container.getT());
+        System.out.println(container.getAnimal());
+        System.out.println(container.getMammal());
+        System.out.println(container.getDog());
+    }
+
+    public static void showDataMammal(GenericContainer<? extends Mammal> container) {
+        System.out.println(container.getT());
+        System.out.println(container.getAnimal());
+        System.out.println(container.getMammal());
+        System.out.println(container.getDog());
+    }
+
+    public static void showDataDog(GenericContainer<? extends Dog> container) {
+        System.out.println(container.getT());
```

Comment:
same as above:
here -> instead of container.getAnimal/getMammal/getDog -> a container.getT().getAnimalVariable() and container.getT().getMammalVariable() and container.getT().getDogVariable()  should be printed out